gvr-unittestutils/src/main/java/org/gearvrf/unittestutils/GVRTestUtils.java

```diff
@@ -154,14 +160,17 @@ public void waitForAssetLoad() {
      * @param frames number of frames to wait for
      */
     public void waitForXFrames(int frames) {
+        mFramesLockDone = false;
         testableMain.notifyAfterXFrames(frames);
         synchronized (xFramesLock) {
-            try {
-                xFramesLock.wait();
-            } catch (InterruptedException e) {
-                Log.e(TAG,"",e);
-                return;
-
+            while(!mFramesLockDone) {
```

Comment:
mFramesLockDone needs to be set from within the synchronized block (per Java memory model).